utils/logging.ts

```diff
@@ -0,0 +1,37 @@
+import { Logger, LogFormatter } from "@aws-lambda-powertools/logger";
+import { LogAttributes, UnformattedAttributes } from "@aws-lambda-powertools/logger/lib/types";
+
+type AtatLog = LogAttributes;
+
+class AtatLogFormatter extends LogFormatter {
+  public formatAttributes(attributes: UnformattedAttributes): AtatLog {
+    return {
+      timestamp: this.formatTimestamp(attributes.timestamp),
+      logLevel: attributes.logLevel,
+      message: attributes.message,
+      region: attributes.awsRegion,
+      correlationIds: {
+        awsRequestId: attributes.lambdaContext?.awsRequestId,
+        xRayTraceId: attributes.xRayTraceId,
+      },
+      service: {
+        name: attributes.serviceName,
+        environment: attributes.environment,
+      },
+      lambdaFunction: {
+        name: attributes.lambdaContext?.functionName,
+        arn: attributes.lambdaContext?.invokedFunctionArn,
+        memoryLimitInMB: attributes.lambdaContext?.memoryLimitInMB,
+        version: attributes.lambdaContext?.functionVersion,
+        coldStart: attributes.lambdaContext?.coldStart,
+      },
+    };
+  }
+}
+
+const baseLogger = new Logger({
+  serviceName: process.env.POWERTOOLS_SERVICE_NAME ?? "ATAT",
+  logFormatter: new AtatLogFormatter(),
+});
+
+export const logger = baseLogger;
```

Comment:
You've taken the logger to the next level, nice!

Reply:
I am really happy with this formatting!! I think eventually we'd be able to pass through the caller identity to the logs too. I've shared a screenshot of what this looks like in a Slack thread. And I think the only real remaining thing to have logging go the extra mile would be `middy` integration and some pre-built CloudWatch queries.